app/controllers/clientDeclarationJourney/ClientResendEmailController.scala

```diff
@@ -0,0 +1,135 @@
+
+        getIntermediaryName(waypoints, request.userAnswers) { intermediaryName =>
+
+          getClientCompanyName(waypoints, request.userAnswers) { clientCompanyName =>
+
+            registrationConnector.getPendingRegistration(uniqueUrlCode).flatMap {
+              case Right(savedPendingRegistration) =>
+
+                sendEmail(
+                  savedPendingRegistration,
+                  clientEmail,
+                  clientCompanyName,
+                  intermediaryName
+                ).flatMap { _ =>
+                  logger.info(s"Successfully resent activation email for uniqueUrlCode: $uniqueUrlCode")
+
+                  for {
+                    updatedAnswers <- Future.fromTry(request.userAnswers.set(EmailWasSentQuery, true))
+                    _ <- sessionRepository.set(updatedAnswers)
+                  } yield {
+                    Redirect(controllers.clientDeclarationJourney.routes.ClientCodeEntryController.onPageLoad(waypoints, uniqueUrlCode))
+                  }
+                }.recover {
+                  case ex =>
+                    logger.error(s"Failed to resend activation email: ${ex.getMessage}", ex)
+                    Redirect(controllers.clientDeclarationJourney.routes.ClientCodeEntryController.onPageLoad(waypoints, uniqueUrlCode))
+                }
+
+              case Left(errorResponse) =>
+                val message = s"Failed to retrieve pending registration for resending email: $errorResponse"
+                logger.error(message)
+                Future.successful(
+                  Redirect(controllers.clientDeclarationJourney.routes.ClientCodeEntryController.onPageLoad(waypoints, uniqueUrlCode))
+                )
+            }
+          }
+        }
+      }
+  }
+  
+  private def sendEmail(
+                         submittedRegistration: SavedPendingRegistration,
+                         clientEmail: String,
+                         clientCompanyName: String,
+                         intermediaryName: String
+                       )(implicit hc: HeaderCarrier, messages: Messages): Future[EmailSendingResult] = {
+    emailService.sendClientActivationEmail(
+      intermediary_name = intermediaryName,
+      recipientName_line1 = clientCompanyName,
+      activation_code_expiry_date = submittedRegistration.activationExpiryDate,
+      activation_code = submittedRegistration.uniqueActivationCode,
+      emailAddress = clientEmail,
+      alertQueue = Some("priority")
+    )
+  }
+  
+  private def getIntermediaryName(waypoints: Waypoints, userAnswers: UserAnswers)(block: String => Future[Result]): Future[Result] = {
+    userAnswers.get(IntermediaryDetailsQuery).map { intermediaryDetails =>
+      block(intermediaryDetails.intermediaryName)
+    }.getOrElse(Redirect(JourneyRecoveryPage.route(waypoints)).toFuture)
+  }
+
+  private def getClientCompanyName(waypoints: Waypoints, userAnswers: UserAnswers)
```

Comment:
I believe there is already functionality available for you to call so this would be duplicated code.